src/commands/cmd_bulk_insert.c

```diff
@@ -57,43 +46,61 @@ void _MGraph_BulkInsert(void *args) {
     RedisModule_ThreadSafeContextLock(ctx);
 
     RedisModuleString **argv = context->argv + 1; // skip "GRAPH.BULK"
-    int argc = context->argc - 2; // skip "GRAPH.BULK [GRAPHNAME]"
     RedisModuleString *rs_graph_name = *argv++;
+    int argc = context->argc - 2; // skip "GRAPH.BULK [GRAPHNAME]"
     RedisModuleKey *key;
-    size_t nodes = 0;   // Number of nodes created.
-    size_t edges = 0;   // Number of edge created.
 
     char reply[1024] = {0}; // Prepare the Redis string response
     int len;
 
     GraphContext *gc = NULL;
 
-    // Type chosen to match Redis conversion function
-    long long final_node_count;
-    long long final_edge_count;
+    // Number of entities already created
+    size_t initial_node_count = 0;
+
+    // Number of entities being created in this query
+    // (declared as long longs to match Redis conversion function)
+    long long nodes_in_query;
+    long long relations_in_query;
+
     // Check the next to see if this is a starting query
+    bool initial_query = false;
     if (!strcmp(RedisModule_StringPtrLen(*argv, 0), "BEGIN")) {
         argv ++;
-        if (_BeginBulkInsert(ctx, rs_graph_name, &final_node_count, &final_edge_count, argv) != BULK_OK) {
+        argc --;
+        initial_query = true;
+        // Verify that graph does not already exist.
+        key = RedisModule_OpenKey(ctx, rs_graph_name, REDISMODULE_READ);
+        if (key) {
+            RedisModule_ReplyWithError(ctx, "Key was already in Redis database at beginning of bulk insert operation.");
```

Comment:
Please change to something similar to:
"Graph with ID: %s can't be created, as Redis key %s already exists"

Reply:
:+1: